sdk/cosmos/azure-cosmos/docs/StatusCodes.md

```diff
@@ -0,0 +1,53 @@
+|410|21006|Yes|`Global strong write barrier not met` - Indicates that synchronous replication of a write operation in a multi-region account with strong consistency did not complete. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|410|21007|Yes|`Read quorum not met` - Indicates that no read quorum could be achieved when using strong or bounded staleness consistency. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|412|0|No|`Precondition failed` - The document has been modified since the application read it (and retrieved the etag that was used as pre-codnition for the write operation). This is the typical optimistic concurrency signal - and needs to be gracefully handled in your application. The usual patterns is to re-read the document, apply the same changes and retry the write with the updated etag. See [Precondition failed TSG - trouble-shooting guide](https://aka.ms/CosmosDB/sql/errors/precondition-failed) for more details.|
+|429|3200|Depends on app RU/s usage|`User throttling` - Indicates that the operations being processed by your Cosmos DB account exceed the provisioned throughput RU/s. Mitigation can be done by either scaling-up - or improving the efficiency especially of queries to reduce the RU/s consumption. See [Throttling TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-request-rate-too-large) for more details.|
+|429|3201|Yes|`Metadata throttling` - Indicates that metadata operations are being throttled. Increasing provisioned throughput (RU/s) won't help - this usually indicates a bug in your application where metadata calls are triggered extensively or you are not using a singleton pattern for `CosmosClient`/`CosmosAsyncClient`. See [Throttling TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-request-rate-too-large) for more details.|
+|429|< 3200|Yes (up to few minutes)|`SLA violating throttling` - Indicates service-side throttling that will count against the service's SLA. These errors should always be transient.| 
+|449|0|Yes|`RetryWith` - Indicates a concurrent attempt to change documents server-side - for example via patch or stored procedure invocation. The `449` status code will be automatically retried by the SDK. This condition should always be transient as long as the application is not excessively doing concurrent changes to documents.|
+|500|0|Unknown|`Internal Server error` - Indicates unexpected and unqualified internal service error.|
```

Comment:
From service telemetry 500 is a small fraction compated to 503, approx 5%.
Also HTTP spec wise 503 is more specific and say that service is unavailable. How about recommended retry on 500 also?

Reply:
Agreed - but that should go into the "building resilient apps" doc - don't want to clutter any instructions when to do app-level retries here. Also called this out explicitly in the overview/out of scope section.